### PR TITLE
add option for metalink directive to yumpkg.

### LINKED ISSUE #58931: pkgrepo.managed on CentOS doesn't support metalink directive
Currently in Salt 3000.3, setting up a repository with pkgrepo.managed on CentOS requires either a `baseurl` or `mirrorlist` directive.  However, some repositories, including EPEL 7 and 8, are now using `metalink` exclusively. 

Suggested functionality:

```
epel-repo:
    pkgrepo.managed:
      - name: epel
      - humanname: Extra Packages for Enterprise Linux 7 - $basearch
      - metalink: https://mirrors.fedoraproject.org/metalink?repo=epel-7&arch=$basearch&infra=$infra&content=$contentdir
      - enabled: 1
      - gpgcheck: 1
      - failovermethod: priority
      - gpgkey: file:///etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-7
```

Attempting to apply this state gives:

```
          ID: epel-repo
    Function: pkgrepo.managed
        Name: epel
      Result: False
     Comment: Failed to configure repo 'epel': The repo does not exist and needs to be created, but either a baseurl or a mirrorlist needs to be given
     Started: 12:38:54.523276
    Duration: 5678.068 ms
     Changes:   
```

### PR DESCRIPTION
**Pull request by racooper**
### What does this PR do?
Adds the ability to use metalink directive for yumpkg repo configuration.

### What issues does this PR fix or reference?
Fixes: #58931

### Previous Behavior
attempting to use metalink directive in yum configs in place of baseurl or mirrorlist failed.

### New Behavior
Allows a yum repo to be configured with either baseurl, mirrorlist or metalink.

### Merge requirements satisfied?
Updated documentation for pkgrepo but don't know anything about writing tests.

### Commits signed with GPG?
No

### CODE REVIEW

File: salt/modules/yumpkg.py [hunk 1]
```diff
@@ -2923,10 +2923,12 @@ def mod_repo(repo, basedir=None, **kwargs):
         the URL for yum to reference
     mirrorlist
         the URL for yum to reference
+    metalink
+        the URL for yum to reference
```
Comment (Ch3LL) [inline salt/modules/yumpkg.py:2927]: can you add `.. versionadded:: 3006` for this new argument.

File: salt/states/pkgrepo.py [hunk 2]
```diff
@@ -212,6 +213,9 @@ def managed(name, ppa=None, copr=None, aptkey=True, **kwargs):
     mirrorlist
         A URL which points to a file containing a collection of baseurls
 
+    metalink
+        A URL for a curated list of non-stale mirrors
```
Comment (Ch3LL) [inline salt/states/pkgrepo.py:217]: can you add .. versionadded:: 3006 for this new argument.

**Review by Ch3LL — CHANGES_REQUESTED** [review]
This will also require a changelog and test coverage

**Comment by racooper** [comment]
This is the first time I've submitted a change, and I don't know anything about writing tests.  Would happy to have some guidance on how to proceed.

File: salt/states/pkgrepo.py (hunk 2 above)
Comment (racooper) [reply salt/states/pkgrepo.py:217]: Added, I think.

File: salt/modules/yumpkg.py (hunk 1 above)
Comment (racooper) [reply salt/modules/yumpkg.py:2927]: Added, I think.

**Review by racooper — COMMENTED** [review]
added versionadded flag, I think in the correct places.

**Comment by Ch3LL** [comment]
@racooper here is our documentation on testing: https://docs.saltproject.io/en/latest/topics/development/tests/index.html

I think a great place to add some test coverage would be in the test files:

tests/pytests/functional/states/pkgrepo/test_centos.py
tests/pytests/unit/modules/test_yumpkg.py

If you also need additional help, there is a test clinic on our twitch stream that occurs often, where you can ask specific test questions. The schedule can be found here: https://www.twitch.tv/saltprojectoss

**Comment by Ch3LL** [comment]
@racooper do you have some time to loop back to this PR and submit some test coverage?

**Comment by Ch3LL** [comment]
bump @racooper are you willing to push this PR to the finish line with adding test coverage and a changelog?

**Comment by racooper** [comment]
I don't have enough experience nor time to learn what I need to finish this up. Is there anyone willing to make the necessary tests for this?  It's becoming more important since EPEL for EL9 is no longer providing a baseurl or mirrorlist URL for their repositories.

**Comment by dwoz** [comment]
Closing this due to inactivity. Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.

**Comment by racooper** [comment]
I do not have the first clue about writing unit tests. I'm not a developer by training or trade, I'm a sysadmin.  I was hoping that someone might be able to pick up what I've done and complete the tests side of things.  Frankly, I'm not even sure what to be testing for, much less how to write it.  I've looked through the pytest files that were previously recommended, but I really have no idea where to start.

**Comment by dwoz** [comment]
@whytewolf Would you like to take a crack at writing tests for this?

**Comment by whytewolf** [comment]
I think i can, but i also want to redo that three tiered all to be a more efficient sum comparison.

**Pull request closed without merging**